src/controllers/teacher.controller.js

```diff
@@ -134,7 +138,41 @@ const getSupplies = async (req, res) => {
             _id: { $in: supplyIds },
         });
         res.status(200).json({
-            supplies: supplies
+            supplies
+        });
+    } catch (err) {
+        return res.status(400).json({
+            error: errorHandler.getErrorMessage(err),
+        });
+    }
+};
+
+const getStudents = async (req, res) => {
+    const teacher = req.profile;
+    const studentIds = teacher.students;
+    try {
+        // Finds all sutudents by id in the students array
+        const students = await Student.find({
+            _id: { $in: studentIds },
+        });
+        // each student has a donations array - want to return 
+        // an array of the donation objects (not donation_ids)
+        // so the supplyItem and quantityDonated 
+        // can be displayed in the teacher dashboard donors table
+        for( let i=0; i<students.length; i++ ){
```

Comment:
<img width="169" alt="CleanShot 2022-11-12 at 23 13 15@2x" src="https://user-images.githubusercontent.com/55270854/201510489-5b773110-c977-499f-806c-c843990caf1c.png">

Since the Donation model stores the student id, I think it'll be easier to do a donations.find({ student: student._id }) and then push that result into an array.  You won't need a nested for loop which should improve performance.

Reply:
Our Donation model has actually be revised to store a reference to the supply using supply_id instead of supply in case the teacher edits the supply item name. The item name is currently not being stored, but only assigned when it's needed in a query. 
![Screen Shot 2022-11-13 at 1 23 10 PM](https://user-images.githubusercontent.com/107423305/201540208-ec6812c0-c5d7-4444-b71c-9b3d34b97917.png)